### ISSUE TITLE
devtool edit-recipe doesn't work

### ISSUE THREAD

**Issue by coldspark29** [garmin/pyrex#72]
``devtool edit-recipe`` doesn't find vi. Quentin already found the root of the cause. The EDITOR env variable needs to be passed to Pyrex and all common editors should be installed in the container.

**Comment by JoshuaWatt** [comment]
We can't run editors in Pyrex; that route leads to user confusion (since it won't be the user's same editor environment will all their bells and whistles), and image bloat (having to include everyone's favorite $EDITOR in the image).

Instead, I think we need to take a different approach and allow some specific commands to break out of the container. This might look like the main pyrex script creating a UNIX domain socket and passing a reference to it inside the container. A shim script in the container would be set as $EDITOR in the container, but would actually pass all it's arguments back to the main pyrex script outside the container over the domain socket; the main pyrex script would execute the correct command and send the result back over the socket.

**Comment by coldspark29** [comment]
Yeah, that sounds like a better solution to me as well, but it will require more work. I was aiming for a quick fix.

**Comment by coldspark29** [comment]
I just had an idea: How if you call ``devtool find-recipe`` when``devtool edit-recipe`` is run and pass back the resulting path to the host with ``| xargs $EDITOR``?
